From 4eb745c8cb5c82a2e01e407648c708de5c5a7d3e Mon Sep 17 00:00:00 2001
From: "Srivatsa S. Bhat" <srivatsa@csail.mit.edu>
Date: Tue, 2 Oct 2018 17:17:36 -0700
Subject: [PATCH] bpf, ext4, bonding: Fix compilation errors
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In file included from kernel/bpf/core.c:24:0:
kernel/bpf/core.c: In function ‘___bpf_prog_run’:
./include/linux/filter.h:750:50: error: cast from function type ‘u64 (*)(u64,  u64,  u64,  u64,  u64) {aka long long unsigned int (*)(long long unsigned int,  long long unsigned int,  long long unsigned int,  long long unsigned int,  long long unsigned int)}’ to non-matching type ‘u64 (*)(u64,  u64,  u64,  u64,  u64,  const struct bpf_insn *) {aka long long unsigned int (*)(long long unsigned int,  long long unsigned int,  long long unsigned int,  long long unsigned int,  long long unsigned int,  const struct bpf_insn *)}’
  ((u64 (*)(u64, u64, u64, u64, u64, const struct bpf_insn *)) \
                                                  ^
kernel/bpf/core.c:1132:13: note: in expansion of macro ‘__bpf_call_base_args’
   BPF_R0 = (__bpf_call_base_args + insn->imm)(BPF_R1, BPF_R2,
             ^~~~~~~~~~~~~~~~~~~~
kernel/bpf/core.c: In function ‘bpf_patch_call_args’:
./include/linux/filter.h:750:50: error: cast from function type ‘u64 (*)(u64,  u64,  u64,  u64,  u64) {aka long long unsigned int (*)(long long unsigned int,  long long unsigned int,  long long unsigned int,  long long unsigned int,  long long unsigned int)}’ to non-matching type ‘u64 (*)(u64,  u64,  u64,  u64,  u64,  const struct bpf_insn *) {aka long long unsigned int (*)(long long unsigned int,  long long unsigned int,  long long unsigned int,  long long unsigned int,  long long unsigned int,  const struct bpf_insn *)}’
  ((u64 (*)(u64, u64, u64, u64, u64, const struct bpf_insn *)) \
                                                  ^
kernel/bpf/core.c:1410:3: note: in expansion of macro ‘__bpf_call_base_args’
   __bpf_call_base_args;
   ^~~~~~~~~~~~~~~~~~~~

Signed-off-by: Srivatsa S. Bhat <srivatsa@csail.mit.edu>
Signed-off-by: Roye Eshed <eshedr@vmware.com>
[Ajay: Dropping include/net/bond_alb.h changes as these
       have been upstreamed]
Signed-off-by: Ajay Kaher <ajay.kaher@broadcom.com>
---
 drivers/gpu/drm/nouveau/Kbuild          | 2 ++
 drivers/gpu/drm/nouveau/dispnv50/wndw.h | 4 ++--
 include/net/bond_alb.h                  | 4 ++--
 4 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/nouveau/Kbuild b/drivers/gpu/drm/nouveau/Kbuild
index b17843d..d9a83ac 100644
--- a/drivers/gpu/drm/nouveau/Kbuild
+++ b/drivers/gpu/drm/nouveau/Kbuild
@@ -62,3 +62,5 @@ nouveau-y += nv84_fence.o
 nouveau-y += nvc0_fence.o
 
 obj-$(CONFIG_DRM_NOUVEAU) += nouveau.o
+
+OBJECT_FILES_NON_STANDARD := y
diff --git a/drivers/gpu/drm/nouveau/dispnv50/wndw.h b/drivers/gpu/drm/nouveau/dispnv50/wndw.h
index b0b6428..0770683 100644
--- a/drivers/gpu/drm/nouveau/dispnv50/wndw.h
+++ b/drivers/gpu/drm/nouveau/dispnv50/wndw.h
@@ -38,8 +38,8 @@ struct nv50_wndw {
 
 int nv50_wndw_new_(const struct nv50_wndw_func *, struct drm_device *,
 		   enum drm_plane_type, const char *name, int index,
-		   const u32 *format, enum nv50_disp_interlock_type,
-		   u32 interlock_data, u32 heads, struct nv50_wndw **);
+		   const u32 *format, u32 heads, enum nv50_disp_interlock_type,
+		   u32 interlock_data, struct nv50_wndw **);
 void nv50_wndw_init(struct nv50_wndw *);
 void nv50_wndw_fini(struct nv50_wndw *);
 void nv50_wndw_flush_set(struct nv50_wndw *, u32 *interlock,
-- 
2.7.4
